From e34e151b59703392227edee6a91732d63a8ca552 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Mon, 16 Aug 2021 10:21:01 -0400
Subject: [PATCH] aufs: update kmalloc_index to v5.14+ restrictions

commit 588c7fa022d7b [mm, slub: change run-time assertion in
kmalloc_index() to compile-time], means that the non-constant
aufs kmalloc call, should go (temporarily) to the __kmalloc_index
directly, so the compile time BUG is not triggered.

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 fs/aufs/module.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/aufs/module.h b/fs/aufs/module.h
index d6a76788d2ff..6c18688d0e3f 100644
--- a/fs/aufs/module.h
+++ b/fs/aufs/module.h
@@ -97,7 +97,7 @@ static inline void au_kfree_small(const void *p)
 static inline int au_kmidx_sub(size_t sz, size_t new_sz)
 {
 #ifndef CONFIG_SLOB
-	return kmalloc_index(sz) - kmalloc_index(new_sz);
+	return __kmalloc_index(sz,false) - __kmalloc_index(new_sz,false);
 #else
 	return -1; /* SLOB is untested */
 #endif
-- 
2.19.1

